astropy/coordinates/representation.py

```diff
@@ -1902,6 +1947,27 @@ def from_cartesian(cls, cart):
         # erfa p2s: P-vector to spherical polar coordinates.
         return cls(*erfa_ufunc.p2s(p), copy=False)
 
+    def transform(self, matrix):
+        r"""Transform the spherical coordinates using a 3x3 matrix.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) `~numpy.ndarray`
+            A 3x3 transformation matrix, such as a rotation matrix.
+
+        """
+        # TODO! differentials
```

Comment:
Could use `self._raise_if_has_differentials('transform')` or perhaps better just run `super().transform` if there are differentials.

Reply:
Doing the latter. 👍